example-app/SantokuApp/src/screens/demo/cache/FileInfo.tsx

```diff
@@ -0,0 +1,87 @@
+import * as FileSystem from 'expo-file-system';
+import React, {useCallback, useEffect, useState} from 'react';
+import {StyleSheet, View} from 'react-native';
+import {Icon, Text} from 'react-native-elements';
+
+import {useCacheDirectory} from './useCacheDirectory';
+
+type FileInfoProps = {
+  fileInfo: FileSystem.FileInfo;
+  currentDepth: number;
+};
+
+const maxDepth = 10;
+
+const FileInfo: React.FC<FileInfoProps> = props => {
+  const {fileInfo, currentDepth} = props;
+  const {readDirectoryItemsFileInfoAsync} = useCacheDirectory();
+  const [childFileInfos, setChildFileInfos] = useState<FileSystem.FileInfo[]>([]);
+
+  const getFileNameFromUri = useCallback((uri: string) => {
+    return uri.endsWith('/') ? uri.split('/').slice(-2)[0] : uri.split('/').slice(-1)[0];
+  }, []);
+
+  useEffect(() => {
+    if (fileInfo.isDirectory && currentDepth <= maxDepth) {
+      readDirectoryItemsFileInfoAsync(fileInfo.uri)
+        .then(fileInfos => {
+          setChildFileInfos(fileInfos);
+        })
+        .catch(e => {
+          console.log(e);
+        });
+    }
+  }, [currentDepth, fileInfo, readDirectoryItemsFileInfoAsync]);
```

Comment:
子要素を表示するかどうかの判定が２か所で必要になっているので、一か所で纏めて計算していいかと思います。

```suggestion
  const isViewChildren = fileInfo.isDirectory && currentDepth <= maxDepth;
  useEffect(() => {
    if (isViewChildren) {
      readDirectoryItemsFileInfoAsync(fileInfo.uri)
        .then(fileInfos => {
          setChildFileInfos(fileInfos);
        })
        .catch(e => {
          console.log(e);
        });
    }
  }, [isViewChildren, fileInfo, readDirectoryItemsFileInfoAsync]);
```